roachtest/awsdms: add large data replication test

Jeremyyang920 opened the pull request:
This commit adds a test that will create a dms task to full load 200M rows into a crdb cluster.

Fixes: #95329
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/96193)
<!-- Reviewable:end -->

Commit:
roachtest/awsdms: add large data replication test

This commit adds a test that will create a dms task
to full load 200M rows into a crdb cluster.

Fixes: #95329
Release note: None

otan commented on pkg/cmd/roachtest/tests/awsdms.go:
```diff
@@ -39,18 +41,24 @@ import (
+	awsdmsCRDBDatabase        = "defaultdb"
+	awsdmsCRDBUser            = "dms"
+	awsdmsNumInitialRows      = 100000
+	awsRDSSecretName          = "rds!db-074de488-6274-4b3e-ad27-d008b8ffa750"
```
how comfortable are we holding this secret information out on an open source repo?

Jeremyyang920 replied:
It should be fine given you need to have an IAM role that can access our AWS account. If someone from outside was able to do that we have problems.

Jeremyyang920 replied:
This password was going to be automatically rotated by secret manager until I learned this morning if you use postgres as a DMS endpoint, your password cant contain +,$ or ; and some of the generated passwords will contain those...

otan replied:
can you a comment to that effect. can you also add a comment detailing that we keep this RDS instance up all the time?

also nit: can you put these `RDS` secrets in a separate const block and rename it to `awsrds...`

otan commented on pkg/cmd/roachtest/tests/awsdms.go:
```diff
@@ -68,13 +76,20 @@ const tableRules = `{
+var dmsEndpoints = make(map[string]string)
 var dmsTasks = []dmsTask{
-	{"test_table", tableRules, nil},
+	{"test_table", tableRules, nil, dmstypes.MigrationTypeValueFullLoadAndCdc, "default_source", "default_target"},
```
can you add field names to all these, e.g. `tableName: "test_table", tableRules: ....`

otan commented on pkg/cmd/roachtest/tests/awsdms.go:
```diff
@@ -203,14 +224,15 @@ func runAWSDMS(ctx context.Context, t test.Test, c cluster.Cluster) {
 	}
 	targetPGConn := c.Conn(ctx, t.L(), 1)
 
 	checkDMSReplicated(ctx, t, sourcePGConn, targetPGConn)
```
can we use `t.Run` on all of these so that they come out as subtests?

Jeremyyang920 replied:
There doesnt appear to be a t.Run method on the t object

otan replied:
forgot to delete this comment

otan commented on pkg/cmd/roachtest/tests/awsdms.go:
```diff
@@ -577,7 +699,7 @@ func setupDMSEndpointsAndTask(
 				ServerName:         rdsCluster.Endpoint,
 			},
-			arn: &sourceARN,
+			key: "default_source",
```
i think these key names might conflict with `{master, release-22.1, ...}` is running at the same time.

Jeremyyang920 replied:
It should still generate the name with the suffix of the release at the end though right?
"roachtest-awsdms-rds-endpoint-" + awsdmsVerString(v)"
So there shouldn't be conflict. The key in this code here is mostly just for the map to the ARN of the endpoint

otan replied:
didn't realise the key wasn't used for RDS identification. gotcha.

otan commented on pkg/cmd/roachtest/tests/awsdms.go:
```diff
@@ -772,11 +907,12 @@ func tearDownDMSTasks(ctx context.Context, t test.Test, dmsCli *dms.Client) erro
 func tearDownDMSEndpoints(ctx context.Context, t test.Test, dmsCli *dms.Client) error {
-	for _, ep := range []string{awsdmsRoachtestDMSRDSEndpointName(t.BuildVersion()), awsdmsRoachtestDMSCRDBEndpointName(t.BuildVersion())} {
+	for _, ep := range dmsEndpoints {
+		fmt.Println(ep)
```
nit: remove

otan commented on pkg/cmd/roachtest/tests/awsdms.go:
```diff
@@ -68,13 +76,20 @@ const tableRules = `{
 	]
 }`
 
+var dmsEndpoints = make(map[string]string)
```
can we do this without a global by e.g. keeping a `state` struct somewhere and passing it through everything?
also how do you feel about this being a struct instead of generic key value

Jeremyyang920 replied:
So the idea with the map was incase we have more sources and targets later on, it maps the name to its arn since to create the task it needs the endpoint arn. But when we create the endpoints, we don't know which tasks are going to use that endpoint so thats why I had like the key: "default_source". So the map is populated and then we look up the arn during the task creation. Can't quite see how I would utilize a struct the best here.

But yes, I can remove the global.

otan replied:
having a static key is basically like having a struct :D so i'd imagine just

```
type dmsEndpointState struct {
   defaultSource string
   defaultTarget string
   bigBoiTarget string
}
```

but up to you.

Jeremyyang920 commented:
bors r=otan
